### DISCUSSION TITLE
Domotiser sa porte de garage pour 5€ - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#6] [Comments]
# Domotiser sa porte de garage pour 5€ - Haade.fr

une façon simple de domotiser sa porte de garage à moindre coût grâce à un module esp8266

https://haade.fr/fr/blog/domotiser-sa-porte-de-garage-pour-5

**Comment by apct69** [comment]
J'ai reçu cette carte relais/ ESP aujourd'hui et codé ce soir! 
En effet l'upload n'est pas gagné, personnellement j'ai téléchargé le fichier binaire généré par esphome sous HA puis flashé grâce à ESPhome Flasher.
Mes leds n'interagissait pas comme expliqué dans le tuto. 
En jouant avec un cavalier sur le boot et en lançant plusieurs fois le flash cela a marché tout seul. 
(indications les bleu et rouge faible m'ont plus induit en erreur qu'autre chose.

J'ai pu ensuite l'intégrer sous HA et upload le code via WIFI en y ajoutant le vôtre. J'ai modifié le code d'entré optocoupleur pour l'utiliser avec un capteur laser pour m'informer de l'ouverture. Pour l'interrupteur physique, je pense le mettre en parallèle du relais et non en série. 
Merci pour les liens ! plus qu'à mettre en place au garage pour ma part :)

**Reply by haade-administrator** [reply]
De rien, ce genre de module domotique peut être capricieux lors du flashage, mais c'est aussi le cas pour d'autres modules esp32 et 8266

**Comment by run51** [comment]
Bonjour

Je voudrais brancher un capteur de position pur confirmer l’ouverture et la fermeture 
Quelle est la modification à faire dans le code?

Merci

**Comment by haade-administrator** [comment]
bonojur pour quel type de capteurs ?

**Comment by run51** [comment]
Pour un capteur magnétique.
Quelle modification je dois faire ?
Le top serait d'avoir un capteur pour l'information porte ouverte et l'autre porte fermée

**Comment by haade-administrator** [comment]
il faut brancher le capteur sur 5v+ et gnd ou si gpio 5 est libre sur gpio/gnd
adapte le code en conséquence
```
binary_sensor:
  - platform: gpio
    name: "Door contact"
    pin:
      number: 5
      inverted: true
      mode:
        input: true
        pullup: true
```

**Comment by run51** [comment]
D'accord merci. Il ne faut pas envoyer du 5v obligatoirement ?Brancher sur le +5 et gpio5?
 
Par contre ce n'est pas possible 2 brancher 2 capteurs ? 1 ouverture et 1 fermeture

Merci

**Reply by run51** [reply]
Bonjour  Je n'ai pas eu de réponse

**Reply by haade-administrator** [reply]
non il n'y a pas plus d'une sortie

**Reply by run51** [reply]
d'accord. Existe t il un modèle avec 2?

**Comment by skatboy** [comment]
Bonjour, est ce que aujourd’hui il serait possible de faire la même chose via ceci : https://haade.fr/fr/blog/test-du-Micro-controlleur-contact-sec-wifi-matter-Sonoff-mini-d 
D’avance merci pour votre retour.
Bonne journée.

**Reply by haade-administrator** [reply]
oui sans problème
